src/js/index.js

```diff
@@ -0,0 +1,62 @@
+import { getSelector } from './../utils';
+import { operator, select } from './../constant';
+import Digit from './digit';
+import Dom from './dom';
+import Calculate from './calculate';
+import './../css/index.css';
+export default class App {
+ constructor($root) {
+  this.dom = new Dom($root, getSelector(select.TOTAL, 'id'));
+  this.number = new Digit();
+  this.operationFn = null;
+ }
+
+ init() {
+  this.dom.setClickEvent(this.handelClick.bind(this));
+ }
+
+ handleEq() {
+  if (!this.operationFn) return;
+  const newNum = this.operationFn(this.number.getNumber());
+
+  this.number = new Digit(newNum.toString());
+  this.dom.print(this.number.getNumber());
+  this.operationFn = null;
+ }
+
+ handelClick(e) {
+  if (!e.target) return;
+  const target = e.target;
+  const classList = target.classList;
+  if (classList.length !== 1) {
+   return;
+  }
+
+  const targetText = target.textContent;
+
+  if (classList.contains(select.OPERATION)) {
+   if (targetText === operator.EQ) {
+    this.handleEq();
+    return;
+   }
+   this.operationFn = new Calculate(this.number.getNumber()).getOperator(
+    targetText
+   );
```

Comment:
```suggestion
    const operators = {
      [operator.PULS]: (prevNumber) => (nextNumber) => prevNumber + nextNumber,
      [operator.MINUS]: (prevNumber) => (nextNumber) => prevNumber - nextNumber,
      [operator.DIVIDE]: (prevNumber) => (nextNumber) => prevNumber / nextNumber,
      [operator.MULTIPLE]: (prevNumber) => (nextNumber) => prevNumber * nextNumber,
    }
   const operation = operators[targetText]; 
   const prevNumber = this.number.getNumber();   
   this.operationFn = operation(prevNumber);
```
(마이너) class로 풀어도 문제가 없긴하지만 단순히 함수를 반환하기 위한 용도라면 이렇게 작성하는게 조금 더 맥락을 이해하는데 쉬울 것 같아요. 
내부적인 필드를 관리하는 케이스가 아니라서 클래스로 굳이 만들어야할지 의문이 들었어요

Reply:
오!! 더 깔끔해지고 좋군요! 구조를 생각할때 객체간의 소통이라고 생각하고 작업하다보니 고민 없이 class 로 구현했던 것 같습니다.

+) operators 객체는 객체리터럴이라서 constant 의 operation.js 파일에 넣을까? 라는 생각도 했었는데 constant 에 있으면 이상할까요?

Reply:
[96001de](https://github.com/next-step/js-calculator/pull/165/commits/96001de7156201af85d28098d9ad3bdf3d8787c4)

Reply:
일반적으로 상수나 메세지등을 constants에 주로 담아두고 있어요.

객체 리터럴이라도 반환하는 용도가 currying 함수다보니 일반적으로 기대하는 위치는 아닐 것 같아요 👀